src/createTwitchVideoCompiler/youtubeUploader.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import fs from 'fs';
+import readline from 'readline';
+import { google } from 'googleapis';
+
+export default async function youtubeUploader(
+  clientFilePath: string,
+  properties: {
+    language: string,
+    description: string,
+    tags: string,
+    title: string,
+    privacy: string,
+    videoPath: string,
+  },
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const { OAuth2 } = google.auth;
+    const SCOPES = ['https://www.googleapis.com/auth/youtube.force-ssl'];
+    // $FlowFixMe
+    const TOKEN_DIR = `${process.env.HOME ||
```

Comment:
We want our code to be as modular and reusable as possible. Using `process.env` here means this function depends on `process` and this dependency could easily be eliminated by getting `TOKEN_DIR` as a parameter.

But, TBH, authentication and uploading are different beasts and we should have two functions, one that does the authentication and another that does uploading.